themes.tf

```diff
@@ -37,36 +37,58 @@ resource "github_repository" "themes" {
   }
 }
 
-resource "github_branch_protection" "themes-main" {
-  repository_id = github_repository.themes.node_id
-
-  pattern                         = "main"
-  enforce_admins                  = false
-  allows_deletions                = false
-  require_signed_commits          = false
-  required_linear_history         = true
-  require_conversation_resolution = true
-  allows_force_pushes             = false
-  lock_branch                     = false
-
-  restrict_pushes {
-    blocks_creations = false
-    push_allowances = [
-      "/${data.github_user.nl-design-system-ci.username}",
-    ]
-  }
-
-  required_status_checks {
-    strict   = false
-    contexts = ["Continuous integration"]
-  }
-
-  required_pull_request_reviews {
-    dismiss_stale_reviews = true
-    restrict_dismissals   = false
-    pull_request_bypassers = [
-      "/${data.github_user.nl-design-system-ci.username}",
-    ]
+resource "github_branch_default" "themes" {
+  repository = github_repository.themes.name
+  branch     = "main"
+}
+
+resource "github_repository_ruleset" "themes-main" {
+  enforcement = "active"
+  name        = "default-branch-protection"
+  repository  = github_repository.themes.name
+  target      = "branch"
+
+  conditions {
+    ref_name {
+      include = ["~DEFAULT_BRANCH"]
+      exclude = []
+    }
+  }
+
+  rules {
+    creation                      = true
+    deletion                      = true
+    non_fast_forward              = true
+    required_linear_history       = true
+    required_signatures           = false
+    update                        = false
+    update_allows_fetch_and_merge = false
+
+    pull_request {
+      dismiss_stale_reviews_on_push     = true
+      require_code_owner_review         = true
+      require_last_push_approval        = false
+      required_approving_review_count   = 1
+      required_review_thread_resolution = true
+    }
+
+    required_status_checks {
+      do_not_enforce_on_create             = false
+      strict_required_status_checks_policy = false
+
+      required_check {
```

Comment:
Waar komt deze nieuwe logica vandaan? Als ik op https://registry.terraform.io/providers/integrations/github/latest/docs/resources/branch_protection#required-status-checks kijk zie ik dat ze nog steeds `contexts` gebruiken? Is dit een nieuwere versie?

Reply:
Oh wacht, het komt doordat je over bent op die ruleset? https://registry.terraform.io/providers/integrations/github/latest/docs/resources/repository_ruleset

Reply:
Dit is https://registry.terraform.io/providers/integrations/github/latest/docs/resources/repository_ruleset 

GitHub wil graag dat je overstapt van wat ze classic branch protection noemen naar ruleset:

<img width="778" alt="screenshot of GitHub wanting you to upgrade to Rulesets" src="https://github.com/user-attachments/assets/0211180b-9ad1-445c-9a6e-f25e5e9e317f" />